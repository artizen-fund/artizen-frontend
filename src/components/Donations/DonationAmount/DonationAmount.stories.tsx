import { useState } from 'react'
import DonationAmount from './'

export default {
  title: 'components/DonationAmount',
  component: DonationAmount,
  argTypes: {},
}

export const DonationAmountComponent = (props: any) => {
  const [amount, setAmount] = useState(10) // note: sort out integer or float
  return <DonationAmount {...props} {...{ amount, setAmount }} />
}
