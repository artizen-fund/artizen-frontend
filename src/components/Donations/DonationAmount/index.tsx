import { useState, useEffect, useContext } from 'react'
import styled from 'styled-components'
import { Button, Icon, AmountWidget, CheckboxControl, DonationHelpLink, Leaderboard } from '@components'
import { breakpoint, palette, typography } from '@theme'
import { rgba, DonationContext } from '@lib'

interface IDonationAmount {
  amount: number
  setAmount: (n: number) => void
  donationMethod: DonationMethod
  setDonationMethod: (d: DonationMethod) => void
}

type MethodSet = {
  key: DonationMethod
  label: string
  min: number
}

const methods: Array<MethodSet> = [
  {
    key: 'usd',
    label: 'Credit Card',
    min: 10,
  },
  {
    key: 'polygon',
    label: 'Polygon',
    min: 10,
  },
  {
    key: 'ethereum',
    label: 'Ethereum',
    min: 100,
  },
]

const DonationAmount = ({ amount, setAmount, donationMethod, setDonationMethod }: IDonationAmount) => {
  const { setDonationStage } = useContext(DonationContext)
  const [hideFromLeaderboard, setHideFromLeaderboard] = useState(false)
  // todo: how is this managed?

  const [minClamp, setMinClamp] = useState(10)
  useEffect(
    () => setMinClamp(methods.find(thisMethod => thisMethod.key === donationMethod)?.min || minClamp),
    [donationMethod],
  )

  return (
    <Wrapper>
      <Information>
        <div>
          <Title>Choose your donation amount and payment method</Title>
          <DonationHelpLink />
        </div>

        <Leaderboard />

        <CheckboxControl
          data={hideFromLeaderboard}
          handleChange={() => setHideFromLeaderboard(!hideFromLeaderboard)}
          label="Hide my personal details from the leaderboard."
          path="derp"
        />
      </Information>
      <Form>
        <AmountWidget {...{ amount, setAmount, minClamp }} />

        <SuggestedDonations>
          <span>Friends of Artizen typically donate:</span>
          <div>
            <Button onClick={() => setAmount(200)} level={2} outline>
              $200
            </Button>
            <Button onClick={() => setAmount(500)} level={2} outline>
              $500
            </Button>
            <Button onClick={() => setAmount(1000)} level={2} outline>
              $1000
            </Button>
          </div>
        </SuggestedDonations>

        <Methods>
          {methods.map(thisMethod => (
            <Method
              key={thisMethod.key}
              onClick={() => setDonationMethod(thisMethod.key)}
              selected={donationMethod === thisMethod.key}
            >
              <Icon outline level={2} glyph="info" />
              <div>{thisMethod.label}</div>
              <span>min ${thisMethod.min}.00</span>
            </Method>
          ))}
        </Methods>

        <Button onClick={() => setDonationStage?.('login')} stretch level={1}>
          Continue
        </Button>
      </Form>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    flex-direction: row;
    gap: 80px;
  }
`

const Information = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

const Title = styled.h1``

const InfoLine = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 15px;
`

const Form = styled.div`
  flex: 1;
`

const SuggestedDonations = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: 1em 0;
  div {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 8px;
  }
`

const Methods = styled.ul`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  gap: 10px;
  margin: 1em 0;
`

const Method = styled.li<{ selected: boolean }>`
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  div {
    margin-top: 0.5em;
    ${typography.label.l1}
  }

  span {
    ${typography.label.l2}
    color: ${rgba(palette.barracuda)};
  }

  border: 0.5px solid ${props => rgba(props.selected ? palette.night : palette.stone)};
  @media (prefers-color-scheme: dark) {
    border: 0.5px solid ${props => rgba(props.selected ? palette.moon : palette.barracuda)};
  }
  transition: border 0.3s ease-in-out;
  border-radius: 16px;
  padding: 10px 0;
  cursor: pointer;
`

export default DonationAmount
