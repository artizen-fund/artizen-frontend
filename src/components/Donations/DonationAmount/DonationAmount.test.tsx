import renderer from 'react-test-renderer'
import DonationAmount from './'

describe('DonationAmount', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DonationAmount />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
