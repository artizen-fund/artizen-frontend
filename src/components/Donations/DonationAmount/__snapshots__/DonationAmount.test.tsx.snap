// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationAmount renders correctly 1`] = `
<div
  className="sc-fa264814-0 kXMtVP"
>
  <div
    className="sc-fa264814-1 cssaZW"
  >
    <div>
      <h1
        className="sc-fa264814-2"
      >
        Choose your donation amount and payment method
      </h1>
      <div
        className="sc-78ba2856-0 fOnoXs"
      >
        <div
          className="sc-c089a03b-2 ckelpU"
        >
          <div
            className="sc-c089a03b-0 hULMLC"
            color="night"
          >
            <div
              className="sc-149d3190-0 jDOEuv"
              color="night"
            />
          </div>
        </div>
        <p
          className="sc-78ba2856-1 jZynMh"
        >
          Need help? Weâ€™ve prepared a useful guide to donating.
        </p>
      </div>
    </div>
    <p>
      todo: insert Leaderboard here
    </p>
    <label
      className="sc-dd35d48e-4 fUZPkX"
    >
      <div
        className="sc-dd35d48e-0 bkkyVw"
      >
        <input
          checked={false}
          className="sc-dd35d48e-1 clTMzK"
          disabled={false}
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <span
          className="sc-dd35d48e-2 dtceeR"
        />
      </div>
      <span
        className="sc-dd35d48e-3 eAaGSv"
      >
        Hide my personal details from the leaderboard.
      </span>
    </label>
  </div>
  <div
    className="sc-fa264814-4 ejyLEh"
  >
    <div
      className="sc-cf7cfc90-0 dmSvkC"
    >
      <div
        className="sc-cf7cfc90-1 fHdqbB"
      >
        <span>
          USD
        </span>
      </div>
      <div
        className="sc-cf7cfc90-3 dwofuU"
      >
        <input
          className="sc-cf7cfc90-2 fzorkd"
          onChange={[Function]}
          type="number"
        />
        <label>
          donation total
        </label>
      </div>
      <div
        className="sc-cf7cfc90-4 dERrdJ"
      />
    </div>
    <div
      className="sc-fa264814-5 gZwxVK"
    >
      <span>
        Friends of Artizen typically donate:
      </span>
      <div>
        <button
          className="sc-2a588df4-2 dorpEK undefined "
          onClick={[Function]}
        >
          <span>
            $200
          </span>
        </button>
        <button
          className="sc-2a588df4-2 dorpEK undefined "
          onClick={[Function]}
        >
          <span>
            $500
          </span>
        </button>
        <button
          className="sc-2a588df4-2 dorpEK undefined "
          onClick={[Function]}
        >
          <span>
            $1000
          </span>
        </button>
      </div>
    </div>
    <ul
      className="sc-fa264814-6 yLhfE"
    >
      <li
        className="sc-fa264814-7 gcZSMc"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="sc-c089a03b-2 ckelpU"
        >
          <div
            className="sc-c089a03b-0 hULMLC"
            color="night"
          >
            <div
              className="sc-149d3190-0 jDOEuv"
              color="night"
            />
          </div>
        </div>
        <div>
          Credit Card
        </div>
        <span>
          min $
          10
          .00
        </span>
      </li>
      <li
        className="sc-fa264814-7 gcZSMc"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="sc-c089a03b-2 ckelpU"
        >
          <div
            className="sc-c089a03b-0 hULMLC"
            color="night"
          >
            <div
              className="sc-149d3190-0 jDOEuv"
              color="night"
            />
          </div>
        </div>
        <div>
          Polygon
        </div>
        <span>
          min $
          10
          .00
        </span>
      </li>
      <li
        className="sc-fa264814-7 gcZSMc"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="sc-c089a03b-2 ckelpU"
        >
          <div
            className="sc-c089a03b-0 hULMLC"
            color="night"
          >
            <div
              className="sc-149d3190-0 jDOEuv"
              color="night"
            />
          </div>
        </div>
        <div>
          Ethereum
        </div>
        <span>
          min $
          100
          .00
        </span>
      </li>
    </ul>
    <button
      className="sc-2a588df4-2 jVJhXF undefined "
      onClick={[Function]}
    >
      <span>
        Continue
      </span>
    </button>
  </div>
</div>
`;
